pkg/virt-api/api.go

```diff
@@ -869,7 +871,15 @@ func (app *virtAPIApp) prepareCertManager() {
 func (app *virtAPIApp) registerValidatingWebhooks(informers *webhooks.Informers) {
 
 	http.HandleFunc(components.VMICreateValidatePath, func(w http.ResponseWriter, r *http.Request) {
-		validating_webhook.ServeVMICreate(w, r, app.clusterConfig)
+		validating_webhook.ServeVMICreate(w, r, app.clusterConfig,
+			func(field *field.Path, vmiSpec *v1.VirtualMachineInstanceSpec, clusterCfg *virtconfig.ClusterConfig) []metav1.StatusCause {
+				netValidator := netadmitter.NewValidator(field, vmiSpec, clusterCfg)
+				var statusCauses []metav1.StatusCause
+				statusCauses = append(statusCauses, netValidator.ValidateCreation()...)
+				statusCauses = append(statusCauses, netValidator.Validate()...)
+				return statusCauses
+			},
```

Comment:
Its possible, but these is a con: it introduce a dependency of `pkg/network/admitters` on `pkg/virt-config`, because the mentioned function (admitters.ValidateSpec) signature specify `*virtconfig.ClusterConfig`.

Introducing a helper and that's signature specify the `clusterConfigChecker` interface instead wont pass compilation, due to mismatch type at the caller (virt-api/api.go).